### PR TITLE
[CLIPPER-102] Integrate threadpool with TaskExecutor, RPC layer

### PR DESCRIPTION
**Pull request by Corey-Zumar**

### CODE REVIEW

File: src/libclipper/include/clipper/task_executor.hpp [hunk 1]
```diff
@@ -101,9 +102,10 @@ class TaskExecutor {
             active_containers_->add_container(
                 vm, std::stoi(container_info["zmq_connection_id"]),
                 replica_id, parse_input_type(container_info["input_type"]));
-            // TODO: This callback should be submitted to a thread pool
-            // for execution on a separate thread
-            on_container_ready(vm, replica_id);
+
+            TaskExecutionThreadPool::submit_job([=]() {
```
Comment (dcrankshaw) [inline src/libclipper/include/clipper/task_executor.hpp]: What is this closure capturing? Can we explicitly capture only the variables we need?

Comment (dcrankshaw) [reply src/libclipper/include/clipper/task_executor.hpp]: I think this can be: `TaskExecutionThreadPool::submit_job([this, vm, replica_id]() {`

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Looks good. Just make the closure capture explicit.

File: src/libclipper/include/clipper/task_executor.hpp (hunk 1 above)
Comment (Corey-Zumar) [reply src/libclipper/include/clipper/task_executor.hpp]: Agreed, it's better to explicitly capture only the required variables. Fixed!

**Review by dcrankshaw — APPROVED** [review]

**Comment by dcrankshaw** [comment]
LGTM

**Pull request merged**